test/test_reductions.py

```diff
@@ -814,6 +814,17 @@ def test_prod(self, device, dtype):
         torch.prod(x, 1, out=res2)
         self.assertEqual(res1, res2)
 
+    def test_prod_bool(self, device):
+        tests = [
+            ([True, True], True),
```

Comment:
And an empty tensor input, too, and instead of comparing with expected create a NumPy array from the inputs and compare with the result of `np.prod`.

Reply:
Addressed! Thanks for suggestion!